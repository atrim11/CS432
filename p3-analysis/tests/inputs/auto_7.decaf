def int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}
int numbers[4];
def void main() {
    int result;
    int a;
    int x;
    // Infinite recursion edge case
    result = fibonacci("string");  // Error: wrong type in recursion

    // Redeclaration of variables
    a = 5;a = 10;  // Error: redeclaration

    // Invalid array access
    x = numbers[5];  // Error: out of bounds

    // Invalid type in if condition
    if (result + 5) {  // Error: non-boolean in 'if'
        print_str("Control error");
    }

    // Variable misuse
    result = a + b;  // Error: 'b' is undeclared

    // Function call errors
    result = fibonacci(fibonacci("text"));  // Error: wrong type in nested call
    result = invalidReturn();  // Error propagating from return mismatch
}
