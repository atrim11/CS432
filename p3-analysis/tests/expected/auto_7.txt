Type mismatch in parameter 0 of call to 'fibonacci': expected int but found str on line 13
Type mismatch: bool expected but int found on line 22
Symbol 'b' undefined on line 27
Type mismatch in parameter 0 of call to 'fibonacci': expected int but found str on line 30
Symbol 'invalidReturn' undefined on line 31
'main' must return an integer
